src/libasr/codegen/asr_to_python.cpp

```diff
@@ -0,0 +1,634 @@
+                throw LCompilersException("Cannot represent the boolean operator as a string");
+            }
+        }
+    }
+
+    template <typename T>
+    void visit_body(const T &x, std::string &r, bool apply_indent=true) {
+        if (apply_indent) {
+            inc_indent();
+        }
+        for (size_t i = 0; i < x.n_body; i++) {
+            visit_stmt(*x.m_body[i]);
+            r += s;
+        }
+        if (apply_indent) {
+            dec_indent();
+        }
+    }
+
+    std::string get_type(const ASR::ttype_t *t) {
+        std::string r = "";
+        switch (t->type) {
+            case ASR::ttypeType::Integer : {
+                r += "i";
+                r += std::to_string(ASRUtils::extract_kind_from_ttype_t(t)*8);
+                break;
+            } case ASR::ttypeType::Real : {
+                r += "f";
+                r += std::to_string(ASRUtils::extract_kind_from_ttype_t(t)*8);
+                break;
+            } case ASR::ttypeType::Complex : {
+                r += "c";
+                r += std::to_string(ASRUtils::extract_kind_from_ttype_t(t)*8);
+                break;
+            } case ASR::ttypeType::Character : {
+                r = "str";
+                break;
+            } case ASR::ttypeType::Logical : {
+                r = "bool";
+                break;
+            } default : {
+                throw LCompilersException("The type `"
+                    + ASRUtils::type_to_str_python(t) + "` is not handled yet");
+            }
+        }
+        return r;
+    }
+
+    void visit_TranslationUnit(const ASR::TranslationUnit_t &x) {
+        std::string r = "";
+
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Module_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+                r += "\n";
+            }
+        }
+
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Function_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+                r += "\n";
+            }
+        }
+
+        // Main program
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Program_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+            }
+        }
+        s = r;
+    }
+
+    void visit_Module(const ASR::Module_t &x) {
+        std::string r;
+
+        for (auto &item : x.m_symtab->get_scope()) {
+            if (is_a<ASR::Function_t>(*item.second)) {
+                visit_symbol(*item.second);
+                r += s;
+            }
+        }
+        s = r;
+    }
+
+    void visit_Function(const ASR::Function_t &x) {
+        // Generate code for the lpython function
+        std::string r;
+        r = "def";
+        r += " ";
+        r.append(x.m_name);
+        r += "(";
+        for (size_t i = 0; i < x.n_args; i++) {
+            visit_expr(*x.m_args[i]);
```

Comment:
We have to specify the datatype of the argument here, right?

Something like:
```python
def f(x: i32):
	pass
```

Reply:
Yes! I'll work on this in a follow-up PR. I added a TODO remark so that we don't miss it in future.